src/data/set/lattice.lean

```diff
@@ -332,6 +332,36 @@ supr_option s
 lemma Inter_option {ι} (s : option ι → set α) : (⋂ o, s o) = s none ∩ ⋂ i, s (some i) :=
 infi_option s
 
+lemma Inter_ite {ι : Type*} (f g : ι → set α) (I : set ι) [decidable_pred (λ x, x ∈ I)] :
+  (⋂ i, if i ∈ I then f i else g i) = (⋂ i ∈ I, f i) ∩ (⋂ i (h : i ∉ I), g i) :=
+begin
+  ext x,
+  simp,
+  split,
+  { intro h,
+    split ;
+    { intros i hi,
+      simpa [hi] using h i } },
+  { rintro ⟨h, h'⟩ i,
+    split_ifs with hi,
+    exacts [h i hi, h' i hi] }
+end
```

Comment:
This generalizes to `dite`:
```suggestion
lemma Inter_dite {ι : Type*} (p : ι → Prop)
  (f : Π i, p i → set α) (g : Π i, ¬p i → set α) [decidable_pred p] :
  (⋂ i, if h : p i then f i h else g i h) = (⋂ i (h : p i), f i h) ∩ (⋂ i (h : ¬ p i), g i h) :=
begin
  ext x,
  simp,
  split,
  { intro h,
    split ;
    { intros i hi,
      simpa [hi] using h i } },
  { rintro ⟨h, h'⟩ i,
    split_ifs with hi,
    exacts [h i hi, h' i hi] }
end

lemma Inter_ite {ι : Type*} (f g : ι → set α) (p : ι → Prop) [decidable_pred p] :
  (⋂ i, if p i then f i else g i) = (⋂ i (h : p i), f i) ∩ (⋂ i (h : ¬ p i), g i) :=
Inter_dite _ _ _
```

Does it generalize to `infi` too?

Reply:
It would be good to add the `Union` version too:
```
lemma Union_dite {ι : Type*} (p : ι → Prop)
  (f : Π i, p i → set α) (g : Π i, ¬p i → set α) [decidable_pred p] :
  (⋃ i, if h : p i then f i h else g i h) = (⋃ i (h : p i), f i h) ∪ (⋃ i (h : ¬ p i), g i h) :=
begin
  ext x,
  simp only [set.mem_Union, set.mem_union_eq],
  split,
  { rintro ⟨i, h⟩,
    split_ifs at h with hp,
    { exact or.inl ⟨i, hp, h⟩, },
    { exact or.inr ⟨i, hp, h⟩, }, },
  { rintro (⟨i, hp, h⟩ | ⟨i, hp, h⟩),
    { exact ⟨i, by simp [hp, h]⟩ },
    { exact ⟨i, by simp [hp, h]⟩ }, }
end
```

Reply:
Edited the above suggestion, I found `supr_sup_eq` eventually.

Reply:
I pushed a commit with the 8 variants.

Reply:
Thanks @jcommelin!